meeteval/viz/__main__.py

```diff
@@ -169,30 +172,46 @@ def index_html(
             try:
                 data = get_data_from_html(html_text)
             except json.JSONDecodeError as e:
-                print(f'Error extracting data from {f}. Skipping. {e}')
+                logging.warning(f'Error extracting data from {f}. Skipping. {e}')
                 continue
+    
+            data['absolute_path'] = f.absolute()
 
-            # Add the file name to the data
-            if copy_files:
-                # Copy the file to the output folder
-                # Create a new folder for each source folder to make sure that
-                # the files are not overwritten
-                f_new = out / str(folder_index) / f.name
-                shutil.copy(f, f_new)
-                data['save_path'] = str(f_new.relative_to(out))
-            else:
-                data['save_path'] = str(f.absolute())
-
-            # Disambiguate system names by appending folder path
-            if 'system_name' in data['info']:
-                data['info']['system_name'] = f'{str(folder)}/{data["info"]["system_name"]}'
-            else:
-                data['info']['system_name'] = str(folder)
             avs.append(data)
+    
+    def resolve_system_names(avs):
+        filenames = [av['absolute_path'].parts for av in avs]
+        prefix = os.path.commonprefix(filenames)
+        for group, grouped_avs in itertools.groupby(
+            avs, 
+            key=lambda x: x['absolute_path'].parts[len(prefix)] 
+                if len(prefix) < len(x['absolute_path'].parts) - 1 
+                else None
+        ):
+            grouped_avs = list(grouped_avs)
+            if group is not None:
+                for av in grouped_avs:
+                    av['info']['system_name'] = f'{group}_{av["info"]["system_name"]}'
+                resolve_system_names(grouped_avs)
```

Comment:
The order is wrong.
It should be
```
    >>> avs = [
    ...     {'absolute_path': Path('/a/b/c/file1.tex'), 'info': {'system_name': 'file1'}},
    ...     {'absolute_path': Path('/a/b/c/file2.tex'), 'info': {'system_name': 'file2'}},
    ...     {'absolute_path': Path('/a/b/d/file3.tex'), 'info': {'system_name': 'file3'}},
    ...     {'absolute_path': Path('/a/b/d/file4.tex'), 'info': {'system_name': 'file4'}},
    ...     {'absolute_path': Path('/a/c/file5.tex'), 'info': {'system_name': 'file5'}},
    ... ]
    >>> resolve_system_names(avs)
    >>> for av in avs:
    ...     print(av['info']['system_name'])
    b_c_file1
    b_c_file2
    b_d_file3
    b_d_file4
    c_file5
```
but generates d_b_file4

Reply:
Done